src/ethereum_test_tools/common/helpers.py

```diff
@@ -25,3 +29,74 @@ def to_hash(input: int | str) -> str:
     if type(input) is int:
         return "0x" + input.to_bytes(32, "big").hex()
     raise Exception("invalid type to convert to hash")
+
+
+@dataclass(kw_only=True)
+class CodeGasMeasure(Code):
```

Comment:
Would you mind expanding upon the motivation for this vs. using `gasUsed` in a tx receipt?

Reply:
I'm also curious how the caller will use this. For instance: does the caller need to manually calculate the overhead costs? Or determine if their code leaves the stack dirty?

Reply:
Yes, right now it is expected that the user provides this analysis manually. Though I would really like for this to be automatic, but I'm not sure if it's possible without having a list of all opcodes and their stack modifications.

This is an example of how I am using it right now in the Warm COINBASE tests:
```
"BALANCE": CodeGasMeasure(
            code=bytes(
                [
                    0x41,  # addr: COINBASE
                    0x31,  # BALANCE
                ]
            ),
            overhead_cost=2,
            extra_stack_items=1,
        ),
```

Here, `CodeGasMeasure` is meant to automate the computation of the gas cost of the `BALANCE` opcode.
`overhead_cost` is the gas cost of the `COINBASE` opcode, which we need to subtract to obtain the expected result in the storage.

`extra_stack_items` is the dangling stack item left by the `BALANCE` opcode, which we need to swap+pop before we do the gas measurement, otherwise the subtract operation ends up subtracting the first gas measurement minus the dangling stack element.

Reply:
Okay I understand, thank you.